lib/explorer/series.ex

```diff
@@ -2616,28 +2644,30 @@ defmodule Explorer.Series do
   """
   @doc type: :element_wise
   @spec subtract(
-          left :: Series.t() | number() | NaiveDateTime.t(),
-          right :: Series.t() | number() | NaiveDateTime.t()
+          left :: Series.t() | number() | Date.t() | NaiveDateTime.t() | Duration.t(),
+          right :: Series.t() | number() | Date.t() | NaiveDateTime.t() | Duration.t()
         ) :: Series.t()
-  def subtract(%NaiveDateTime{} = left, %Series{dtype: {:datetime, timeunit}} = right),
-    do: apply_series_list(:subtract, [from_list([left], dtype: {:datetime, timeunit}), right])
-
-  def subtract(%Series{dtype: {:datetime, timeunit}} = left, %NaiveDateTime{} = right),
-    do: apply_series_list(:subtract, [left, from_list([right], dtype: {:datetime, timeunit})])
-
-  def subtract(%NaiveDateTime{} = left, %Series{dtype: {:duration, timeunit}} = right),
-    do: apply_series_list(:subtract, [from_list([left], dtype: {:datetime, timeunit}), right])
-
-  def subtract(%Series{dtype: {:datetime, _}} = left, %Series{dtype: {:datetime, _}} = right),
-    do: apply_series_list(:subtract, enforce_highest_precision(left, right))
+  def subtract(left, right) do
+    [left, right] = cast_for_arithmetic("subtract/2", [left, right])
 
-  def subtract(%Series{dtype: {:datetime, _}} = left, %Series{dtype: {:duration, _}} = right),
-    do: apply_series_list(:subtract, enforce_highest_precision(left, right))
-
-  def subtract(%Series{dtype: {:duration, _}} = left, %Series{dtype: {:duration, _}} = right),
-    do: apply_series_list(:subtract, enforce_highest_precision(left, right))
+    if _dtype = cast_to_subtract(dtype(left), dtype(right)) do
+      apply_series_list(:subtract, [left, right])
+    else
+      dtype_mismatch_error("subtract/2", left, right)
+    end
+  end
 
-  def subtract(left, right), do: basic_numeric_operation(:subtract, left, right)
+  @doc false
+  def cast_to_subtract(:integer, :integer), do: :integer
```

Comment:
We should avoid `@doc false` functions because if the user does `import Explorer.Series`, they will be imported into their namespace. One option is to underscore them, but my suggestion is to move them to `Explorer.Shared`.

Reply:
Addressed by: [093e9ce](https://github.com/elixir-explorer/explorer/pull/696/commits/093e9cea167c7418707e81032b2dd754ac32afc3)